src/solvers/mod.rs

```diff
@@ -199,6 +199,12 @@ pub trait SolverModel {
     fn name() -> &'static str;
 }
 
+/// A solver that can take an initial solution to a problem before solving it
+pub trait WithInitialSolution {
+    /// Sets the initial solution to the problem
+    fn with_initial_solution(self, solution: &Vec<(Variable, f64)>) -> Self;
+}
```

Comment:
Maybe this ?

```suggestion
pub trait WithInitialSolution {
    /// Sets the initial solution to the problem
    fn with_initial_solution(
        self,
        solution: impl IntoIterator<Item = (Variable, f64)>
    ) -> Self;
}
```

Reply:
I had the exact same thought when I woke up today